src/main/kotlin/systems/ajax/malov/stockanalyzer/service/impl/StockRecordRecordAggregationServiceImpl.kt

```diff
@@ -15,16 +15,21 @@ class StockRecordRecordAggregationServiceImpl(
 ) :
     StockRecordAggregationService {
 
-    @SchedulerLock(name = "aggregateStockRecords", lockAtLeastFor = "PT3M", lockAtMostFor = "PT5M")
+    @SchedulerLock(name = "aggregateStockRecords", lockAtLeastFor = "PT50S", lockAtMostFor = "PT3M")
     @Scheduled(cron = "0 0/1 * * * ?")
     override fun aggregateStockRecords() {
-        stockRecordClientApi.run {
-            log.info("Performing aggregation task")
-            getAllStockRecords()
-        }.also {
-            stockRecordRepository.insertAll(it)
-            log.info("Aggregation task finished")
-        }
+        log.info("Performing aggregation task")
+
+        stockRecordClientApi.getAllStockRecords()
+            .collectList()
+            .flatMapMany { records ->
+                stockRecordRepository.insertAll(records)
+            }
+            .then()
+            .doOnSuccess {
+                log.info("Aggregation task finished")
+            }
+            .subscribe()
```

Comment:
Please play with this code:

1. change scheduling to every 5 sec
2. modify your pipeline with emulating long running: `.doOnNext { Thread.sleep(10_000) }`

My expectation: if the previous `aggregateStockRecords` run is not completed then the new one won't start

Reply:
Changed to blocking code to support proper locking